loader/loader_linux.c

```diff
@@ -0,0 +1,635 @@
+                if (is_vulkan_1_1 && device_is_1_1_capable) {
+                    icd_term->dispatch.GetPhysicalDeviceProperties2(sorted_device_info[index].physical_device, &dev_props2);
+                } else {
+                    icd_term->dispatch.GetPhysicalDeviceProperties2KHR(sorted_device_info[index].physical_device, &dev_props2);
+                }
+                sorted_device_info[index].pci_domain = pci_props.pciDomain;
+                sorted_device_info[index].pci_bus = pci_props.pciBus;
+                sorted_device_info[index].pci_device = pci_props.pciDevice;
+                sorted_device_info[index].pci_function = pci_props.pciFunction;
+            }
+            loader_log(inst, VULKAN_LOADER_INFO_BIT | VULKAN_LOADER_DRIVER_BIT, 0, "           [%u] %s", index,
+                       sorted_device_info[index].device_name);
+            index++;
+        }
+    }
+
+    // Select default device in the following order:
+    //   1. If device select environment variable defined, use that
+    //   2. If DRI Prime device select environment variable defined, use that
+    //   3. If XCB is enabled, try selecting default XCB device.
+    //   4. If VGA boot device defined, select that
+    if (linux_env_var_default_device(inst, inst->total_gpu_count, sorted_device_info)) {
+        // Print out selected
+    } else if (linux_dri_prime_default_device(inst, inst->total_gpu_count, sorted_device_info)) {
+        // Print out selected
+    }
+#ifdef VK_USE_PLATFORM_XCB_KHR
+    else if (inst->wsi_xcb_surface_enabled && linux_find_xcb_default_device(inst, inst->total_gpu_count, sorted_device_info)) {
+        // Print out selected
+    }
+#endif
+    else if (linux_vga_boot_default_device(inst, inst->total_gpu_count, sorted_device_info)) {
+        // Print out selected
+    }
+
+    // Sort devices by PCI info
+    qsort(sorted_device_info, inst->total_gpu_count, sizeof(struct LinuxSortedDeviceInfo), compare_devices);
+
+    // If we have a selected index, add that first.
+    loader_log(inst, VULKAN_LOADER_INFO_BIT | VULKAN_LOADER_DRIVER_BIT, 0, "linux_read_sorted_physical_devices:  Order set to:");
+
+    // Add all others after (they've already been sorted)
+    for (uint32_t dev = 0; dev < inst->total_gpu_count; ++dev) {
+        sorted_device_term[dev]->this_icd_term = sorted_device_info[dev].icd_term;
+        sorted_device_term[dev]->icd_index = sorted_device_info[dev].icd_index;
+        sorted_device_term[dev]->phys_dev = sorted_device_info[dev].physical_device;
+        loader_set_dispatch((void *)sorted_device_term[dev], inst->disp);
+        loader_log(inst, VULKAN_LOADER_INFO_BIT | VULKAN_LOADER_DRIVER_BIT, 0, "           [%u] %s", dev,
+                   sorted_device_info[dev].device_name);
+    }
+
+    if (NULL != sorted_device_info) {
+        loader_instance_heap_free(inst, sorted_device_info);
+    }
+
+    return res;
+}
+
+extern uint32_t g_loader_debug;
```

Comment:
This should be a function, `get_loader_debug` whose interface lives in the log header and definition just returns the value of the global variable (thats local to that translation unit).

Death to globals, and where they make sense, encapsulate them so users can't change it without the owners permission.